com.microsoft.mrtk.core/Subsystems/Speech/SpeechRecognitionEventArgs.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.MixedReality.Toolkit.Subsystems
+{
+    /// <summary>
+    /// Event data associated with the result of speech recognition.
+    /// </summary>
+    public class SpeechRecognitionResultEventArgs
+    {
+        /// <summary>
+        /// The result of speech recognition.
+        /// </summary>
+        public string Result { get; private set; }
+
+        /// <summary>
+        /// The confidence of the speech recognition result on a scale of 0 to 1.
+        /// Null when the confidence is not available.
+        /// </summary>
+        public float? Confidence { get; private set; }
+
+        /// <summary>
+        /// Construct the <c>SpeechRecognitionResultEventArgs</c>.
+        /// </summary>
+        public SpeechRecognitionResultEventArgs(string result, float? confidence)
+        {
+            Confidence = confidence;
+            Result = result;
+        }
+    }
+
+    /// <summary>
+    ///  Event data associated with the session of speech recognition.
+    /// </summary>
+    public class SpeechRecognitionSessionEventArgs
+    {
+        /// <summary>
+        /// The reason for this speech recognition session event.
+        /// </summary>
+        public string Reason { get; private set; }
```

Comment:
This feels very loosely-typed; why is this just a string? As a developer I don't know if I would know what to do with this.

Would rather have an extensible enum here.

Reply:
Well, as you might have guessed the underlying implementations use enums, but those are implementation specific. I think we could build an abstraction over those, but part of me really don't want to use enums here as there isn't an "extensible enum" in C# right? And thus future changes to the enum could be breaking.

Reply:
Do we expect the reasons to need to be extensible? I can see the value of the strings for displaying them to a user, but agree with @Zee2 re: using these as a developer. String matching can be very challenging especially if they get localized,

Reply:
As we support more underlying implementations, correctly mapping the implementation-specific reasons to the ones we create could become challenging. Also, I don't really anticipate the reason to be localized. I am open to create an enum and go from there for now, but not sure whether that would open up potential breaking changes down the road.

Reply:
Added `SpeechRecognitionEventReason` enum